codingtest_ajw/16일차/신규_아이디_추천.js

```diff
@@ -0,0 +1,24 @@
+function solution(new_id) {
+    let answer = '';
+    let step1 = /[A-Z]/g;  //대문자 찾기
+    let step2 = /[^a-z0-9-_.]/g; //소문자, 숫자, 빼기, 밑줄, 마침표를 제외한 문자 제거
+    let step3 = /\.+/g;  //마침표 2개 이상이면 1개로 고정
+    let step4_1 = /^[\.]/g; //첫 번째 마침표 찾기
+    let step4_2And6 = /[\.]$/g; //마지막 마침표 찾기
+    let step5 = /\s/g;  //공백 찾기
+    let step7 = /.$/g; //마지막 문자 찾기
```

Comment:
```suggestion
    let step1 = /[A-Z]/g,  //대문자 찾기
      step2 = /[^a-z0-9-_.]/g, //소문자, 숫자, 빼기, 밑줄, 마침표를 제외한 문자 제거
      step3 = /\.+/g,  //마침표 2개 이상이면 1개로 고정
      step4_1 = /^[\.]/g, //첫 번째 마침표 찾기
      step4_2And6 = /[\.]$/g, //마지막 마침표 찾기
      step5 = /\s/g,  //공백 찾기
      step7 = /.$/g; //마지막 문자 찾기
```
let을 한 번만 사용하고 선언이 가능합니다.